### PR TITLE
LG-2711: menu darkMode focus fixes

### PR DESCRIPTION
**Pull request by shaneeza**
This was a quick fix to fix focus styles in darkMode, however, I think this components needs some refactoring. It was very confusing to figure out what styles were doing what and I think a few of the selectors are incorrect.
Screen Shot 2022-12-15 at 12 04 53 PM
Screen Shot 2022-12-15 at 12 04 45 PM
Screen Shot 2022-12-15 at 12 06 08 PM
Screen Shot 2022-12-15 at 12 05 37 PM

## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2711](https://jira.mongodb.org/browse/LG-2711)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/menu/src/MenuItem.tsx [hunk 1]
```diff
@@ -192,6 +191,7 @@ const MenuItem = React.forwardRef(
           {...anchorProps}
           {...rest}
           className={cx(
+            MenuItemContainerClassName,
```
Comment (brookescarlett) [inline packages/menu/src/MenuItem.tsx:194]: nit but should this be `menuItemContainerClassName`

File: packages/menu/src/SubMenu.tsx [hunk 2]
```diff
@@ -529,7 +517,7 @@ const SubMenu = React.forwardRef(
           {content}
           <IconButton
             data-testid="lg-sub-menu-icon-button"
-            darkMode={true}
+            darkMode={!darkMode}
```
Comment (brookescarlett) [inline packages/menu/src/SubMenu.tsx:520]: is IconButton darkmode broken?

File: packages/menu/src/MenuItem.tsx (hunk 1 above)
Comment (shaneeza) [reply packages/menu/src/MenuItem.tsx:194]: oh good catch

File: packages/menu/src/SubMenu.tsx (hunk 2 above)
Comment (shaneeza) [reply packages/menu/src/SubMenu.tsx:520]: I don't think so, we have to use the inverse since the menu becomes light in darkMode.

Comment (shaneeza) [reply packages/menu/src/SubMenu.tsx:520]: oh you mean why we have to pass darkMode?

**Comment by sooachung** [comment]
can we have blue dark 1 for the focus state icon in the light menu and blue light 3 for the dark menu?

File: packages/menu/src/MenuContext.tsx [hunk 3]
```diff
@@ -4,10 +4,12 @@ import { Theme } from '@leafygreen-ui/lib';
 
 interface MenuData {
   theme: Theme;
+  darkMode: boolean;
 }
 
 export const MenuContext = createContext<MenuData>({
   theme: Theme.Light,
+  darkMode: false,
```
Comment (brookescarlett) [inline packages/menu/src/MenuContext.tsx:12]: is it dangerous to set a default here? will we overwrite a value that comes in from the LG provider?

File: packages/menu/src/MenuItem.tsx [hunk 4]
```diff
@@ -192,6 +191,7 @@ const MenuItem = React.forwardRef(
           {...anchorProps}
           {...rest}
           className={cx(
+            menuItemContainerClassName,
```
Comment (brookescarlett) [inline packages/menu/src/MenuItem.tsx:194]: why are we changing where this is defined?

File: packages/menu/src/MenuContext.tsx (hunk 3 above)
Comment (shaneeza) [reply packages/menu/src/MenuContext.tsx:12]: I don't think so, TS complains if nothing is set but it is overwritten when we pass in a value.

File: packages/menu/src/styles.ts [hunk 5]
```diff
@@ -311,7 +311,7 @@ export const focusedMenuItemContainerStyle: Record<Theme, string> = {
     &:focus {
       text-decoration: none;
       background-color: ${palette.blue.light2};
-      color: ${palette.white};
```
Comment (brookescarlett) [inline packages/menu/src/styles.ts:314]: was darkmode broken previously?

Comment (shaneeza) [reply packages/menu/src/styles.ts:314]: its hard to say, I worked on darkMode a few months ago and i remember going through all of this and had design QA everything so i'm not sure what happened

File: packages/menu/src/MenuItem.tsx (hunk 4 above)
Comment (shaneeza) [reply packages/menu/src/MenuItem.tsx:194]: This was initially on this element but was moved to the glyph in this [PR](https://github.com/mongodb/leafygreen-ui/commit/f2d63a60ed0d8669f44fd8e6f45af827b875414f#diff-3f4788e0b950199ad21de29f52ccc19e90bf4acf14da596954b54da22dc966f6L134). Also, sased on the name, it's suppose to be on the menu item container. I think we need to spend some time cleaning up this component.

Comment (brookescarlett) [reply packages/menu/src/MenuItem.tsx:194]: 👍

File: packages/menu/src/styles.ts (hunk 5 above)
Comment (brookescarlett) [reply packages/menu/src/styles.ts:314]: kk

**Comment by shaneeza** [comment]
> can we have blue dark 1 for the focus state icon in the light menu and blue light 3 for the dark menu?

Screen Shot 2022-12-15 at 1 01 54 PM
Screen Shot 2022-12-15 at 1 00 22 PM
@sooachung is this correct?

**Comment by shaneeza** [comment]
> can we have blue dark 1 for the focus state icon in the light menu and blue light 3 for the dark menu?
Screen Shot 2022-12-15 at 2 38 44 PM
Screen Shot 2022-12-15 at 2 38 52 PM
@sooachung

**Comment by sooachung** [comment]
Sorry for not catching it but can the focus state text in the dark menu be light 3 if it is white?
Screen Shot 2022-12-15 at 3 07 51 PM

**Comment by shaneeza** [comment]
> Sorry for not catching it earlier but can the focus state text in the dark menu be light 3 if it is white? Screen Shot 2022-12-15 at 3 07 51 PM
Screen Shot 2022-12-15 at 3 16 55 PM

**Comment by sooachung** [comment]
LGTM!

**Review by sooachung — APPROVED** [review]

**Comment by sooachung** [comment]
Actually two more! 
Can we make this icon grey base?
Screen Shot 2022-12-15 at 3 26 16 PM

and this text grey dark 1?
Screen Shot 2022-12-15 at 3 30 06 PM

**Comment by shaneeza** [comment]
> 



> Actually two more! Can we make this icon grey base? Screen Shot 2022-12-15 at 3 26 16 PM
> 
> and this text grey base? Screen Shot 2022-12-15 at 3 30 06 PM

Screen Shot 2022-12-15 at 4 18 02 PM
Screen Shot 2022-12-15 at 4 16 57 PM

**Comment by shaneeza** [comment]
@sooachung ^^

**Review by sooachung — APPROVED** [review]

File: packages/menu/src/SubMenu.tsx [hunk 6]
```diff
@@ -44,8 +45,9 @@ import {
 } from './styles';
 import { Size } from './types';
 
-const SubMenuContainerClassName = createUniqueClassName('sub-menu-container');
-const IconButtonClassName = createUniqueClassName('icon-button');
+const subMenuContainerClassName = createUniqueClassName('sub-menu-container');
```
Comment (brookescarlett) [inline packages/menu/src/SubMenu.tsx:48]: non blocking but does it make more sense to define these in the styles files?

**Review by brookescarlett — APPROVED** [review]

File: packages/menu/src/SubMenu.tsx (hunk 6 above)
Comment (shaneeza) [reply packages/menu/src/SubMenu.tsx:48]: This component isn't using the new file structure so when we make that switch we can move it. Right now all the styles are in this same file.

**Pull request merged**